src/Lyn.Protocol/Bolt3/Secret/ISecretProvider.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Lyn.Protocol.Bolt3.Secret
+{
+    public interface ISecretProvider
```

Comment:
I think the name shouldn't be provider, it generates a new secret right?

Reply:
No its the nodes global secret, perhaps a SecretStore is more correct